packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.native.test.tsx

```diff
@@ -0,0 +1,196 @@
+    const onChange = jest.fn();
+    const userPassword = 'Divyanshu$123';
+
+    const { getByPlaceholderText } = renderWithTheme(
+      <PasswordField
+        label="Enter password"
+        placeholder={placeholder}
+        name="password"
+        onChange={onChange}
+      />,
+    );
+
+    const input = getByPlaceholderText(placeholder);
+    fireEvent.changeText(input, userPassword);
+
+    // changeText changes entire text at once
+    expect(onChange).toHaveBeenCalledTimes(1);
+    expect(onChange).toHaveBeenCalledWith({ name: 'password', value: userPassword });
+  });
+
+  it('should handle onBlur', () => {
+    const placeholder = 'Password';
+    const onBlur = jest.fn();
+    const userPassword = 'Divyanshu$123';
+
+    const { getByPlaceholderText } = renderWithTheme(
+      <PasswordField
+        label="Enter password"
+        placeholder={placeholder}
+        name="password"
+        onBlur={onBlur}
+      />,
+    );
+
+    const input = getByPlaceholderText(placeholder);
+
+    // We can't rely on onBlur because it doesn't pass the actual value (just React Native things)
+    fireEvent(input, 'onEndEditing', { nativeEvent: { text: userPassword } });
+    expect(onBlur).toHaveBeenCalledTimes(1);
+    expect(onBlur).toHaveBeenCalledWith({ name: 'password', value: userPassword });
+  });
+
+  /**
+   * No tests for uncontrolled input because react-native-testing-library doesn't support it
+   * https://github.com/callstack/react-native-testing-library/issues/978#issuecomment-1203256954
+   */
+  it('should set value as a controlled input', () => {
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu$123';
+
+    const ControlledInputExample = (): ReactElement => {
+      const [value, setValue] = useState<string | undefined>(valueInitial);
+
+      return (
+        <PasswordField
+          label="Enter password"
+          value={value}
+          onChange={({ value }) => setValue(value)}
+        />
+      );
+    };
+
+    const { getByDisplayValue } = renderWithTheme(<ControlledInputExample />);
+
+    const input = getByDisplayValue(valueInitial);
+
+    fireEvent.changeText(input, valueFinal);
+
+    getByDisplayValue(valueFinal);
+  });
+
+  it('should toggle password visibility on clicking reveal button', () => {
+    const placeholder = 'Password';
+    const userPassword = 'Divyanshu$123';
+
+    const { getByPlaceholderText, getByRole, getByLabelText } = renderWithTheme(
+      <PasswordField
+        label="Enter password"
+        placeholder={placeholder}
+        defaultValue={userPassword}
+      />,
+    );
+
+    const input = getByPlaceholderText(placeholder);
+    const button = getByRole('button');
+
+    // @ts-expect-error TS typings not being picked from library
+    expect(input).toHaveProp('secureTextEntry', true);
+    getByLabelText('Show password');
+
+    fireEvent.press(button);
+
+    // @ts-expect-error TS typings not being picked from library
```

Comment:
Maybe let's create an issue for this?

Reply:
Created https://github.com/razorpay/blade/issues/720